nvi-rest/src/test/java/no/sikt/nva/nvi/rest/create/CreateNoteHandlerTest.java

```diff
@@ -60,14 +73,38 @@ void shouldReturnUnauthorizedWhenMissingAccessRights() throws IOException {
     @Test
     void shouldAddNoteToCandidateWhenNoteIsValid() throws IOException {
         var theNote = "The note";
-        DbCandidate dbCandidate = randomCandidate();
-        Candidate candidate = nviCandidateRepository.create(dbCandidate, List.of(randomApprovalStatus()));
+        var dbCandidate = randomApplicableCandidate();
+        var candidate = nviCandidateRepository.create(dbCandidate, List.of(randomApprovalStatus()));
+        nviService.createPeriod(periodFromCandidate(candidate));
         handler.handleRequest(createRequest(candidate.identifier(), new NviNoteRequest(theNote)), output, context);
         var gatewayResponse = GatewayResponse.fromOutputStream(output, CandidateResponse.class);
         var body = gatewayResponse.getBodyObject(CandidateResponse.class);
         assertThat(body.notes().get(0).text(), is(equalTo(theNote)));
     }
 
+    @Test
+    void shouldReturnBadRequestWhenCreatingNoteAfterReportingPeriodHasBeenClosed()
+        throws IOException {
+        var theNote = "The note";
+        var reportingYear = "2100";
+        mockPeriod(reportingYear);
+        var request = createRequest(UUID.randomUUID(), new NviNoteRequest(theNote));
+        handler.handleRequest(request, output, context);
+        var response = GatewayResponse.fromOutputStream(output, Problem.class);
+
+        assertThat(response.getStatusCode(), is(equalTo(HttpURLConnection.HTTP_BAD_REQUEST)));
```

Comment:
I think 409 Conflict is a better status code to return. 400 means the server is not able to understand the request, but the case here is that the server state does not allow the request.